src/Data/TaskMap.hs

```diff
@@ -99,3 +103,74 @@ memo tm@(TaskMap var) k task = do
             return (m', a)
         (Just !a) -> return (m, a)
     wait a
+
+-- -------------------------------------------------------------------------- --
+-- Batch Tasks
+
+newtype NoBatchResultException k = NoBatchResultException k
+    deriving (Show, Eq, Ord, Generic)
+
+instance (Typeable k, Show k) => Exception (NoBatchResultException k)
+
+-- | Schedule a task that computes a batch of task map entries, that can be
+-- awaited as indiviual tasks.
+--
+-- This function immediately returns the 'Async' handle to the assynchronous
+-- task.
+--
+memoBatchAsync
+    :: Eq k
+    => Hashable k
+    => Typeable k
+    => Show k
+    => TaskMap k v
+    -> [k]
+    -> ([k] -> IO (HM.HashMap k v))
+        -- ^ an action that is used to produce the value if the key isn't in the map.
+    -> IO (Async (HM.HashMap k v))
+memoBatchAsync tm@(TaskMap var) ks batchTask = do
+    modifyMVarMasked var $ \m -> do
+        let missing = filter (not . (`HM.member` m)) ks
+        t <- asyncWithUnmask $ \umask -> umask (batchTask missing)
```

Comment:
Why use `WithUnmask` here if you immediately unmask the entire block being passed? What is the benefit?